### PR TITLE
port tests to 4.08

### PR DESCRIPTION
**Pull request by rgrinberg**
The changes to promotion are looking suspicious. This needs investigation before it is merged.

### CODE REVIEW

**Comment by rgrinberg** [comment]
I've added a commit that invalidates equal timestamps when they are equal to the current time. @aalekseyev or @diml, is this what was meant?

> it's slightly different: you do the check when recording the "old" timestamp instead of when comparing it to the new one

I don't fully understand the comment above from slack.

**Comment by diml** [comment]
Yh, that something like this. However, using the current time is not super reliable as it might not correspond to the time of the filesystem. We looked at how git/hg handle this with @aalekseyev and the best way to do that is to touch a file and look at its timestamp.

@aalekseyev started working on a patch where, when we save the digest DB we throw out all entries whose timestamp is equal to the one observed on the FS at this point.

In the meantime, I suggest to disable the problematic tests on OSX.

**Comment by rgrinberg** [comment]
Sure, will do that.

**Comment by diml** [comment]
PR looks good. Just regarding the matrix of tests, currently we have:

- Build: 4.02 to 4.06
- Test: 4.08 twice
- Build_macos: 4.02 to 4.07

Given that we are going to need OCaml 4.08 to build dune moving forward, it doesn't seem useful to test the build on older versions of OCaml.

I suggest to reduce the matrix to just:

- Test: 4.08, 4.09
- Build_macos: 4.08, 4.09
 
and maybe add trunk later as well

**Comment by aalekseyev** [comment]
The patch Jeremie mentioned is now in https://github.com/ocaml/dune/pull/2478.

**Comment by rgrinberg** [comment]
I'll wait for that PR to be merged and rebase this PR. Hopefully I'll be able to run the tests on OSX.

File: .travis-ci.sh [hunk 1]
```diff
@@ -116,8 +115,8 @@ case "$TARGET" in
     if [ $WITH_OPAM -eq 1 ] ; then
       cat $RUNTEST_NO_DEPS;
       _boot/install/default/bin/dune runtest && \
-      _boot/install/default/bin/dune build @test/blackbox-tests/runtest-js && \
-      _boot/install/default/bin/dune build @test/blackbox-tests/runtest-coq && \
+      # _boot/install/default/bin/dune build @test/blackbox-tests/runtest-js && \
+      # _boot/install/default/bin/dune build @test/blackbox-tests/runtest-coq && \
```
Comment (ejgallego) [inline .travis-ci.sh]: I didn't notice this; as of today the whole coq test suite is broken, including a critical regression in #3191 

What should we do?

**Pull request merged**